### ISSUE TITLE
[websockets.exceptions.InvalidStatusCode: server rejected WebSocket connection: HTTP 403]

### ISSUE THREAD

**Issue by abuyusif01** [gradio-app/gradio#6605]
### Describe the bug

am using gradio_client 0.7.0

take a look at the screenshot below, I tracked the authentication module and the only conclusion i can come up with is:
there's a bug somewhere in it. as u can see when i exit right after the post request in _login() the cookies values are valid (i tested this in browser, U can see the cookies in browser window are thesame as the ones in vscode terminal.) and ofcourse i got an empty json when the auth credentials are wrong. this make sense. But the moment i uncomment the exit and continue executing the module. i got a `{'details': 'Not Authenticated'}` when using h2ogpt_gradio utils. and i got `[websockets.exceptions.InvalidStatusCode: server rejected WebSocket connection: HTTP 403]` using pure gradio_client

### Have you searched existing issues?  🔎

- [x] I have searched and found no existing issues

### Reproduction

try to authenticate a user with h2ogpt where --auth option is enabled.

### Screenshot

![bugs_edited](https://github.com/gradio-app/gradio/assets/49121684/d4614827-decb-4687-b427-159a38340f3d)

### System Info

```shell
# OS: Ubuntu 22.04.3 LTS x86_64
# Host: g5.4xlarge
# Kernel: 6.2.0-1016-aws
# CPU: AMD EPYC 7R32 (16) @ 3.304GHz
# GPU: NVIDIA A10G
# Memory: 7351MiB / 63604MiB

ubuntu@b9c173f8ab79:~$ gradio environment
Gradio Environment Information:
------------------------------
Operating System: Linux
gradio version: 3.50.2
gradio_client version: 0.6.1

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 5.2.0
fastapi: 0.104.1
ffmpy: 0.3.1
gradio-client==0.6.1 is not installed.
httpx: 0.25.2
huggingface-hub: 0.16.4
importlib-resources: 6.1.1
jinja2: 3.1.2
markupsafe: 2.1.3
matplotlib: 3.7.1
numpy: 1.22.0
orjson: 3.9.10
packaging: 23.2
pandas: 1.5.3
pillow: 9.5.0
pydantic: 1.10.13
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0.1
requests: 2.31.0
semantic-version: 2.10.0
typing-extensions: 4.8.0
uvicorn: 0.24.0.post1
websockets: 11.0.3
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.10.0
httpx: 0.25.2
huggingface-hub: 0.16.4
packaging: 23.2
requests: 2.31.0
typing-extensions: 4.8.0
websockets: 11.0.3
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
We're working on fixing the latter issue here: https://github.com/gradio-app/gradio/pull/6556

**Comment by abidlabs** [comment]
Should be fixed in the Gradio 4.9.0 or later @abuyusif01!

**Comment by bibo7086** [comment]
I am facing the same issue while using the Gradio client to connect to a Gradio application running at `http://localhost:7860/`. The client initially reports successful loading of the API, but attempts to establish a WebSocket connection are rejected by the server with an HTTP 403 (Forbidden) status code.

**Error Message:**

```
Loaded as API: http://localhost:7860/ ✔
GR job failed: server rejected WebSocket connection: HTTP 403
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/gradio_client/client.py", line 798, in _inner
    predictions = _predict(*data)
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/gradio_client/client.py", line 827, in _predict
    result = utils.synchronize_async(self._ws_fn, data, hash_data, helper)
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/gradio_client/utils.py", line 540, in synchronize_async
    return fsspec.asyn.sync(fsspec.asyn.get_loop(), func, *args, **kwargs)  # type: ignore
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/fsspec/asyn.py", line 103, in sync
    raise return_result
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/fsspec/asyn.py", line 56, in _runner
    result[0] = await coro
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/gradio_client/client.py", line 1055, in _ws_fn
    async with websockets.connect(  # type: ignore
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/websockets/legacy/client.py", line 637, in __aenter__
    return await self
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/websockets/legacy/client.py", line 655, in __await_impl_timeout__
    return await self.__await_impl__()
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/websockets/legacy/client.py", line 662, in __await_impl__
    await protocol.handshake(
File "/path/to/miniconda3/envs/gradioclient/lib/python3.10/site-packages/websockets/legacy/client.py", line 329, in handshake
    raise InvalidStatusCode(status_code, response_headers)

Loaded as API: http://localhost:7860/ ✔
GR job failed again: server rejected WebSocket connection: HTTP 403
    ... (same stack trace as above)
```

**Steps to Reproduce:**

1. python generate.py --base_model=TheBloke/Mistral-7B-Instruct-v0.2-GGUF --prompt_type=mistral --max_seq_len=4096
2. Run the Gradio client code. 
```
import time
import os
import sys

from gradio_utils.grclient import GradioClient

# self-contained example used for readme, to be copied to README_CLIENT.md if changed, setting local_server = True at first
# The grclient.py file can be copied from h2ogpt repo and used with local gradio_client for example use

if local_server:
    client = GradioClient("http://0.0.0.0:7860")
else:
    h2ogpt_key = os.getenv('H2OGPT_KEY') or os.getenv('H2OGPT_H2OGPT_KEY')
    if h2ogpt_key is None:
        sys.exit("API key not found. Exiting.")
    # if you have API key for public instance:
    client = GradioClient("https://gpt.h2o.ai", h2ogpt_key=h2ogpt_key)


print(client.question("Who are you?"))
```

* **Environment:**
    * Operating System: Ubuntu 22.04.4 LTS
    * gradio: 4.26.0
    * gradio_client:  0.6.1
    
    
 * **Note:**
I suspect this is due to internal network restrictions and using 127.0.0.1 should ideally work. Unfortunately it doesn't. I have tried with multiple versions of Gradio and the Gradio client supported by h2ogpt, but to no avail.
